plugins/auth-node/api-report.md

```diff
@@ -43,6 +44,10 @@ export function getBearerTokenFromAuthorizationHeader(
 
 // @public
 export interface IdentityApi {
+  getIdentity(
+    options: Omit<IdentityApiGetIdentityRequest, 'optional'>,
+  ): Promise<BackstageIdentityResponse>;
+  // (undocumented)
```

Comment:
@Rugvip I tried the overload that you mentioned here https://github.com/backstage/backstage/pull/14533#discussion_r1040172973
Not sure I was successful, haha so please check if this is what you wanted to achieve with it.

Reply:
Not quite as I don't think this'd catch `optional: false`. Cleanest way is probably a type parameter for the request.

```ts
interface IdentityApiGetIdentityRequest<TOptional extends boolean = boolean> {
  optional: TOptional,
  ...
}

getIdentity(options: IdentityApiGetIdentityRequest<true>): Promise<BackstageIdentityResponse | undefined>;
getIdentity(options: IdentityApiGetIdentityRequest): Promise<BackstageIdentityResponse>;
```